src/drivers/nuopc/cime_flds/esmFlds.F90

```diff
@@ -1970,577 +1441,563 @@ subroutine esmFlds_Init(gcomp, rc)
-          call shr_nuopc_fldList_AddFld(fldListTo(compatm)%flds, trim(fldname), &
-               merge_from1=complnd, merge_field1=trim(fldname), merge_type1='copy')
-          call shr_nuopc_fldList_AddMap(fldListFr(complnd)%flds(n1), complnd, compatm, mapconsf, 'one', lnd2atm_smapname)
-       enddo
-    endif
-    call seq_drydep_init() ! TODO: remove this when all components call it separately
+    do num = 1, max_ddep
+       write(cnum,'(i2.2)') num
+       fldname  = 'Sl_dd' // cnum
+       call shr_nuopc_fldList_AddFld(fldListFr(complnd)%flds, trim(fldname), fldindex=n1)
+       call shr_nuopc_fldList_AddMap(fldListFr(complnd)%flds(n1), complnd, compatm, mapconsf, 'one', lnd2atm_smapname)
+       call shr_nuopc_fldList_AddFld(fldListTo(compatm)%flds, trim(fldname), &
+            merge_from1=complnd, merge_field1=trim(fldname), merge_type1='copy')
+    end do
 
     !-----------------------------------------------------------------------------
     ! Nitrogen Deposition fields
-    ! First read namelist and figure out the ndepdep field list to pass
-    ! Then check if file exists and if not, n_drydep will be zero
-    ! Then add nitrogen deposition fields to atm export, lnd import and ocn import
     !-----------------------------------------------------------------------------
 
-    call shr_ndep_readnl("drv_flds_in", ndep_fields, ndep_nflds)
-    if (ndep_nflds > 0) then
-       longname = 'nitrogen deposition flux'
-       stdname  = 'nitrogen_deposition'
-       units    = 'kg(N)/m2/sec'
-       do n = 1,ndep_nflds
-          call shr_string_listGetName(ndep_fields, n, fldname)
-          call shr_nuopc_fldList_AddMetadata(fldname=trim(fldname), longname=longname, stdname=stdname, units=units)
-          call shr_nuopc_fldList_AddFld(fldListFr(compatm)%flds, trim(fldname), fldindex=n1)
-          call shr_nuopc_fldList_AddFld(fldListTo(complnd)%flds, trim(fldname), &
-               merge_from1=compatm, merge_field1=trim(fldname), merge_type1='copy_with_weights', merge_fracname1='ofrac')
-          call shr_nuopc_fldList_AddFld(fldListTo(compocn)%flds, trim(fldname), &
-               merge_from1=compatm, merge_field1=trim(fldname), merge_type1='copy_with_weights', merge_fracname1='ofrac')
-          call shr_nuopc_fldList_AddMap(fldListFr(compatm)%flds(n1), compatm, complnd, mapbilnr, 'one', atm2lnd_smapname)
-          call shr_nuopc_fldList_AddMap(fldListFr(compatm)%flds(n1), compatm, compocn, mapbilnr, 'one', atm2ocn_smapname)
-       enddo
-    end if
+    call shr_nuopc_fldList_AddFld(fldListFr(compatm)%flds, 'Faxa_noy', fldindex=n1)
+    call shr_nuopc_fldList_AddMap(fldListFr(compatm)%flds(n1), compatm, complnd, mapbilnr, 'one', atm2lnd_smapname)
+    call shr_nuopc_fldList_AddMap(fldListFr(compatm)%flds(n1), compatm, compocn, mapbilnr, 'one', atm2ocn_smapname)
+    call shr_nuopc_fldList_AddFld(fldListTo(complnd)%flds, 'Faxa_noy', &
+         merge_from1=
```

Comment:
Clearly a step in the right direction - but what else can we do to reduce the size of this file and reuse more code?